compiler/types.nim

```diff
@@ -1632,6 +1635,23 @@ proc isTupleRecursive*(t: PType): bool =
   var cycleDetector = initIntSet()
   isTupleRecursive(t, cycleDetector)
 
+proc isRecursivePointer(typ: PType, isPointer: bool): bool =
+  if typ == nil:
+    return false
+  case typ.kind:
+    of tyObject, tyTuple:
+      return isPointer and canFormAcycle(typ)
+    of tyRef, tyPtr:
+      return isRecursivePointer(typ.lastSon, isPointer = true)
```

Comment:
Infinite recursoin for the illegal type ``T = ptr T``. I think this can matter for ``nim check``.

Reply:
Is there a tag for illegal types that I can just check?

Reply:
I'm not sure, ideally such a type would be a ``tyError``.

Reply:
I will add a test for such a type. If illegal types are correctly marked as ``tyError``, then this shouldn't be an infinite recursion.